x-pack/plugins/ingest_pipelines/public/application/services/navigation.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const BASE_PATH = '/';
+
+const EDIT_PATH = 'edit';
+
+const CREATE_PATH = 'create';
+
+const getEditPath = (name: string, encode = true): string => {
+  return `${BASE_PATH}${EDIT_PATH}/${encode ? encodeURIComponent(name) : name}`;
+};
+
+const getCreatePath = (): string => {
+  return `${BASE_PATH}${CREATE_PATH}`;
+};
+
+const getClonePath = (name: string, encode = true): string => {
```

Comment:
I think we can improve the readability of the code by using config parameters for these path factories. Consider the difference:

```js
// current
getListPath(updatedPipeline.name);
getClonePath(':sourceName', false);

// my suggestion
getListPath({ inspectedPipeline: updatedPipeline.name });
getClonePath({ clonedPipeline: ':sourceName', skipEncode: true });
```

I think it generally makes sense to avoid config parameters when you're only passing in one or two arguments, but I think in this situation it's very helpful for communicating the intention of the code.